testing stuff

KodeRad opened the pull request:
## Summary

Summarize your PR. If it involves visual changes include a screenshot or gif.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

KodeRad committed:
testing stuff

semd committed:
[One Workflow] Reuse rule registry alert formatting (#246219)

## Summary

Addresses: https://github.com/elastic/kibana/issues/243225

No changes in the behaviour, it's just reusing the actual `formatAlerts`
function from rules, instead of replicating the logic in the workflows
plugin.

This approach has an additional advantage (besides removing code
duplication), which is that it uses the right formatter for each alert,
retrieving it from the rule registry, so it would work even for
non-security alerts (if we want to query them in the future).

<img width="1273" height="543" alt="Captura de pantalla 2025-12-12 a les
16 09 37"
src="https://github.com/user-attachments/assets/cd829e66-0d1d-4d20-ac5c-9a7efa0e59ca"
/>

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

criamico committed:
[Fleet] Hide agent csv export in serverless (#246586)

Related to  https://github.com/elastic/kibana/issues/232193

## Summary
Hide csv export menu action in serverless while a definitive solution to
the export problem is worked out.


### Checklist
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

alexwizp committed:
Address `@elastic/eui/require-table-caption` lint violations across `@elastic/kibana-core` files (#246014)

> [!CAUTION]
> ⚠️ **Changes / translations were made by GenAI**. I’ve reviewed them
carefully, but your code owners’ expert eyes will ensure they’re 100%
right.

## Summary
This PR applies the auto-fix for the newly introduced
`@elastic/eui/require-table-caption`.
This rule ensure `EuiInMemoryTable`, `EuiBasicTable` have a
`tableCaption` prop for accessibility.

## Changes

1. 🎯 Added missing `tableCaption` attributes to elements flagged by
`@elastic/eui/require-table-caption` — accessibility leveled up!

## Related
- https://github.com/elastic/eui/pull/9168

This time, to avoid annoying approvals collection, we've broken files
down by teams. Now, we are waiting a review only from your team!

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Gerard Soldevila <gerard.soldevila@elastic.co>

cauemarcondes committed:
[Discover][Traces] Adding error marks to unified trace waterfall (#245161)

kibanamachine commented:
<!-- kbn-message-context:project-deploy-job -->
PR Project deployment started at: https://buildkite.com/elastic/kibana-deploy-project-from-pr/builds/817

bartoval committed:
[ES|QL] Show next actions after simple field assignment in RERANK ON Clause (#246676)

_## Summary

There is an incorrect interpretation of the RERANK grammar, namely
`rerankField qualifiedName (ASSIGN BooleanExpression)`. This does not
mean it should return a Boolean, but simply the name describing the
rule.

so even simple fields are valid

jennypavlova committed:
[Metrics in Discover] Unskip metrics api test (#246593)

Closes [#244709](https://github.com/elastic/kibana/issues/244709)

## Summary

This PR unskips the failed test, which was fixed after the issue skipped
it. The test passed multiple times locally.

### Checklist

- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed:
[Pased](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/10112)
✔️

gsoldevila committed:
Add missing `dynamic: false` (#246685)

## Summary

It should be the default behavior for all SO types!

bartoval committed:
[ES|QL  ]  Common Lookup Join Fields Are Not Listed First (#246582)

## Summary

Added two new sorting categories for common and unique lookup fields. 

Common has higher priority than fields and unique lookup fields (same
priority).

AlexGPlay committed:
[Background search] Change polling behavior (#244760)

## Summary

Closes https://github.com/elastic/kibana/issues/230909

Right now after a search has finished we keep polling it every 30
seconds, this was because search sessions allowed to save them after
they finished, but with background search that's not the case, so we
don't need to poll.

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

KodeRad committed:
[AI Infra] Chat experience: Selection modal title change (#246683)

## Summary

It changes the text of the Selection modal to "Select a chat experience"

<img width="1009" height="621" alt="Screenshot 2025-12-17 at 11 03 05"
src="https://github.com/user-attachments/assets/f801d042-8040-4c46-9602-f4cbf87d02ac"
/>




### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

ppisljar committed:
[agent builder] create_visualization: support heatmap and regionmap (#246671)

Commit:
[o11y AI] Add `get_hosts` tool (#246541)

Adds the `observability.get_hosts` tool, enabling agents to retrieve
infrastructure host metrics.

**Example**
```
POST kbn://api/agent_builder/tools/_execute
{
  "tool_id": "observability.get_hosts",
  "tool_params": {
    "start": "now-4h",
    "end": "now"
  }
}
```

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

mykolaharmash committed:
[Streams] Add simulation filtering by conditions (#245400)

Closes https://github.com/elastic/kibana/issues/239909

🔒
[Figma](https://www.figma.com/design/C4o0y9Knk4jYXjsulvbW3w/Streams-Processors--Grok?m=auto&node-id=9552-33019&t=8ZoAb4lgexLLXxBg-1)

## Summary
This change adds an option to focus the simulation on a specific WHERE
condition by clicking on it or using the condition's context menu.


https://github.com/user-attachments/assets/6e443865-a679-46e1-b2cd-fec01a113655

## Key implementation details
* `_simulate` endpoint was modified to return a new `processed_by`
property on every document that went through simulation. This property
holds a list of processor IDs that affected a specific document.
* Using the new `processed_by` property, client code is able to filter
documents in the table when user clicks on a condition by collecting all
processors that are included in the selected condition (and implicitly
previous processors as they could be affecting the partial simulation)
and mapping them to the documents with `processed_by`.
* The documents are filtered on the client side which give a nice user
experience of an immediate feedback. Additionally, a partial simulation
is executed in the background in order to update the relative percentage
numbers for the table filters and the step list.
* A `baseSimulation` property was added to the simulation state machine
context in order to hold the full simulation containing all sample
documents. This is needed for cases when user switches between
conditions without clearing the previously condition selection, each
condition needs to have access to the full simulation in order to see
full documents list with `processed_by` properties.

> [!IMPORTANT]  
> Using the described approach with the `processed_by` property gives a
few key benefits:
> * It enables the client side filtering with immediate feedback
> * It avoids complicating the `_simulate` endpoint logic and keeps it
almost unchanged. If we would go with server side filtering, `_simulate`
endpoint would need to be aware of the concept of filtering by condition
in order to properly calculate the metrics for only the subset of the
documents.
> 
> At the same time, because we have no way to faithfully evaluate WHERE
conditions without running some special conditions-only simulation,
filtering of the documents is based on the processors (both included in
the target condition and implicitly processors from the previous steps)
and not strictly on the condition expression. This might produce
unexpected results in terms of visible documents for cases when, for
example, processors themselves include a condition which makes them
hidden from the table even though user selected a condition that matches
those documents. While technically correct, users might have questions.
For most users though, I expect the current UX to be transparent enough.

## How to test
1. Run Kibana with some data flowing into Streams
2. Create a processing steps with some conditions
3. Click on one of the conditions or use "Preview this only" item inside
its context menu
4. Check that documents left in the table correspond to the processors
you have inside the condition
5. Check that percentages displayed on the condition's steps and on the
table filters are correct

robester0403 committed:
Add Canonical Name Badge to Documentation (#246647)

Adds an indication in integrations' documentation of the integration's
canonical identifier to reduce incorrect reporting of integration names
and so assignments when support cases are raised.

cnasikas committed:
[ResponseOps][Slack] Simplify channel configuration  (#245423)

## Summary

Fixes: https://github.com/elastic/kibana/issues/230979


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

kubasobon committed:
[Entity Store][API] Fix snake case on CRUD API List response (#246003)

## Summary

Closes https://github.com/elastic/kibana/issues/241120

A quick fix for `entity.attibutes`, `entity.behaviors`, and
`entity.lifecycle` entries being returned with ECS formatting. Now they
are lower snake case, as expected.

<img width="1998" height="928" alt="image"
src="https://github.com/user-attachments/assets/fa781005-740e-4b68-9171-19cf34aae5f2"
/>

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

dennis-tismenko committed:
[Agent Builder] Fast-follow bugfixes for MCP Tool type  (#246665)

## Summary

A couple of relatively small bugfixes for MCP tool type support in Agent
Builder bundled together:
- __MCP tool search not working on the bulk import page__: I forgot that
if you have a custom change handler for `search`, you need to handle the
filtering as well
- __Select all MCP tools__: Would inadvertently get reset when clicking
on the "x selected" button due to some reconciliation logic built into
`EuiInMemoryTable`
- __Namespace validation__: Focused on the complicated one first but
forgot to add the simple ones — added similar validation rules to tool
IDs
- __UX improvement: Automatically select newly created connector__
- __Health status improvements__: Disabled retries for health checks
(faster feedback) and some minor refactor and improvements to how
they're rendered


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

tonyghiani committed:
[Streams] Test enable wired streams flow (#246113)

## 📓 Summary

Related to https://github.com/elastic/streams-program/issues/573

Implement tests to verify that enabling wired streams with the UI works
as expected.

---------

Co-authored-by: Robert Stelmach <60304951+rStelmach@users.noreply.github.com>
Co-authored-by: Robert Stelmach <robert.stelmach@elastic.co>

KodeRad committed:
add ff

gsoldevila committed:
Retry on "all shards failed" from ES (#246533)

## Summary

We are observing Pod restarts on Serverless during Kibana bootstrap
(migration) process, with the following error:

```
Reason: Unable to complete saved object migrations for the [.kibana] index. Error: pickupUpdatedMappings task failed with the following error:
{"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":".kibana_1","node":"iu4ANm6gTQO8eQyvnWGdkw","reason":{"type":"no_shard_available_action_exception","reason":"[es-es-search-6b8cf96d57-rvnz4][100.66.190.214:9300][indices:data/read/search[phase/query]]"}}]}
```

It seems that the Kibana Pods bootstrap normally after the restart, so
we can assume that the `all shards failed` is a temporary condition of
ES upon which we can retry.

spong committed:
[Agent Builder] Adds keyboard shortcut and toggle behavior to AI Agent button (#246659)

## Summary

Adds keyboard shortcut and toggle behavior to `AI Agent` button.

Press `cmd ;` on Mac or `ctrl ;` on Windows to toggle visibility of the
flyout (just as before with the O11y/Security Assistants). Functions in
each Solution, Stack Management, and regardless of currently focused
component.


https://github.com/user-attachments/assets/f0d2aa37-eb27-4361-b992-19558a1ee95e


### Checklist

Check the PR satisfies following conditions. 




Reviewers should verify this PR satisfies this list as well.

- [X] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios


_PR developed with Cursor + GPT 5.2_

vadimkibana committed:
[ES|QL] PromQL parser initial implementation (#246552)

## Summary

Partially addresses https://github.com/elastic/kibana/issues/243932

This PR adds PromQL language parsing support to `kbn-esql-ast`. The
parser is mostly complete with one known TODO in the CST-to-AST
converter for `SubqueryResolutionContext` and `NonReservedContext`
handling. Also, there might be some AST interface tweaks going forward,
but overall the parser works.

Note: The PromQL ANTLR grammar we use, which is ported from
Elasticsearch repo, does not exactly match the Prometheus PromQL
language (some language features are missing, some implemented slightly
differently).

The parser allows to provide `offset` option, which bumps all
`.location` positions of AST nodes by that value. This will allow us to
parse nested PromQL queries within ES|QL and set their `.location`
correctly.

The PromQL AST nodes contain an extra `dialect: 'promql'` field, which
helps to differentiate them from the ES|QL nodes.

Most AST nodes re-use the existing node shape from the ES|QL AST nodes
(e.g. numbers, string, identifiers), however, other are specific for
PromQL (function with grouping, selector, etc..), so they have their own
interface and node type.

### Testing

```sh
yarn test:jest src/platform/packages/shared/kbn-esql-ast/src/promql
```


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: Stratou <stratoula1@gmail.com>

christineweng committed:
[Cases] Change nested field search to be case insensitive (#246643)

## Summary

Updated observables value and custom field search to be case
insensitive. This would match how title, description and comments are
currently being searched.


Video demoing case insensitivity for title, comments and observables.


https://github.com/user-attachments/assets/75ed64f4-d6f6-419e-a967-8a9f8dd9c245


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

sbelastic committed:
[APM] Migrate the Transaction Overview tests to Scout/Playwright/Component/API tests (#245972)

Closes https://github.com/elastic/kibana/issues/238515

## Summary

Migrating transactions overview tests from Cypress to Scout

### Running tests e2e

Run server (first terminal):

`node scripts/scout.js start-server --stateful`

Run tests (second terminal):

`
npx playwright test --project local --ui --config
x-pack/solutions/observability/plugins/apm/test/scout/ui/parallel.playwright.config.ts
`


### Running tests api

Run server (first terminal):

`node scripts/functional_tests_server --config
x-pack/solutions/observability/test/api_integration_deployment_agnostic/configs/stateful/oblt.apm.stateful.config.ts`

Run tests (second terminal):

`node scripts/functional_test_runner --config
x-pack/solutions/observability/test/api_integration_deployment_agnostic/configs/stateful/oblt.apm.stateful.config.ts
--include=x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/apm/transactions/transactions_groups_order.spec.ts`

### New tests result output

<img width="1919" height="1122" alt="image"
src="https://github.com/user-attachments/assets/7167375c-879d-410a-912a-ab6581011fdf"
/>

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.


- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

mbondyra committed:
[CPS][Maps] Support CPS Picker in Maps  (#246382)

## Summary

Supports CPS project picker in Maps. When changing the picker setting,
maps refetch the data with the correct `project_routing` parameter.
*Note*: This does NOT allow yet saving the project routing setting to
the maps saved object

Also fixes the fact I forgot to pass `project_routing` to esql source.

#### To test: 
1. Navigate to Maps and create a new map
2. Add a few layers with Elasticsearch data sources (e.g., documents,
clusters, heatmap)
3. Open browser DevTools Network tab (optionally, filter for search
requests)
  4. Click the CPS picker and change the project routing setting
  5. Verify that:
    - Maps layers refetch data automatically
- Search requests include `project_routing` parameter in the request
body
- Different CPS settings result in different project_routing values in
requests (none for All projects)


https://github.com/user-attachments/assets/8cbd4348-e2b0-445b-b260-64d5b1a3077f

nchaulet committed:
[Fleet] Do not allow to edit agentless agent policy (#246439)

tara-elastic committed:
implement lorana's suggestions (#246427)

## Summary

- Additional clarification on Notion-connector documentation
- Clean up some redundant/unnecessary tests


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ]  #...

resolves elastic/search-team#12103

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

Bluefinger committed:
[Discover] Toggle Icon type for add/remove field column (#246024)

## Summary

Closes https://github.com/elastic/kibana/issues/238802


https://github.com/user-attachments/assets/795a8dc6-3db9-4560-8f54-b82d74a437f5

This PR adds a toggle behaviour to the icon for the toggle column
action. When the field is present as a column, it shows as a minus, and
when it is not present, it shows as a plus. This behaviour is also
present for the Logs Overview message actions.

## How to test

* Go to Discover with logs index showing, select a log to open up the
Overview.
* Go to either the Attributes tab, or the Table tab, and find a field to
add to the data-grid.
* Toggle the field, so it shows on the data-grid, and then remove it.
The toggle column icon should update according to the columns state.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine commented:
<!-- kbn-message-context:project-deploy-job -->
Project deployed, see credentials at: https://buildkite.com/elastic/kibana-deploy-project-from-pr/builds/818

ersin-erdal committed:
Validate worker inputs in PDF Generation Process (#245330)

This PR adds an input validation to the PDF generation worker to block
malicious data.

The worker is used only for **non-printable** PDF generation.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

CoenWarmer committed:
[Streams] No longer use spread as it spread `stepId` onto the component (#246722)

## Summary

This fixes a React error being output to the browser console in dev
mode.

<img width="494" height="308" alt="Screenshot 2025-12-17 at 13 29 03"
src="https://github.com/user-attachments/assets/9aa58c4b-8bb9-4566-abe8-56e8b459a9b6"
/>

React flags camel cased props that are output to DOM elements. In this
case, due to spreading of all props, `stepId` was output to the DOM.

juliaElastic committed:
[Fleet] fix deprecated ilm check (#246718)

## Summary

Relates https://github.com/elastic/kibana/pull/243416

It seems `filter_path` doesn't work as expected, there are no component
templates returned.
Without it, the Deprecated ILM policies warning is showing up as
expected.

- first modify a `@package` component template to use a deprecated
lifecycle e.g. `logs`
- go to Fleet / Data streams and check that the warning callout is
visible

<img width="1635" height="1067" alt="image"
src="https://github.com/user-attachments/assets/a0f363c1-e393-4430-bd23-583ddbd167f4"
/>

<img width="1632" height="721" alt="image"
src="https://github.com/user-attachments/assets/6b32316e-df11-460c-ba0e-d5126637bfc8"
/>


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

MiloszRadzynski committed:
[Discover][Flaky Test Fix] Close datepicker popover by clicking escape key (#246698)

## Summary

This PR adds one step in the `setAbsoluteRange` function, which is
closing the datepicker popover via escape key. It makes sure that the
popover does not cover the actual datepicker buttons which could
potentially cause the test flakiness.

It closes: #246229
it closes: #246228
it closes: #246468

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

KodeRad committed:
fix ce visibility on serverless
